gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -138,7 +146,8 @@ public <V> void onRetry(Attempt<V> attempt) {
             }
           }
         }));
-        this.eventProducer = observabilityEventProducer;
+    this.eventProducer = observabilityEventProducer;
+    this.dagProcessingEngine = dagProcessingEngine;
```

Comment:
Why are we interacting with `dagProcessingEngine`? We should be passing work to `newDagManager`